API/Repositories/Implementations/UserRepository.cs

```diff
@@ -7,46 +7,28 @@ namespace API.Repositories.Implementations;
 
 [SuppressMessage("Performance", "CA1862:Use the \'StringComparison\' method overloads to perform case-insensitive string comparisons")]
 [SuppressMessage("ReSharper", "SpecifyStringComparison")]
-public class UserRepository(ILogger<UserRepository> logger, OtrContext context) : RepositoryBase<User>(context), IUserRepository
+public class UserRepository(OtrContext context) : RepositoryBase<User>(context), IUserRepository
 {
-    private readonly ILogger<UserRepository> _logger = logger;
     private readonly OtrContext _context = context;
 
     public override async Task<User?> GetAsync(int id) =>
         await UserBaseQuery().FirstOrDefaultAsync(u => u.Id == id);
 
-    public async Task<User?> GetForPlayerAsync(int playerId) =>
-        await UserBaseQuery().FirstOrDefaultAsync(u => u.PlayerId == playerId);
+    public async Task<User?> GetByOsuIdAsync(int osuId) =>
```

Comment:
`osuId` should always be passed as `long`, make sure this is being done everywhere.